src/ir/value.h

```diff
@@ -0,0 +1,114 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+#ifndef SRC_IR_VALUE_H_
+#define SRC_IR_VALUE_H_
+
+#include <vector>
+
+#include "absl/container/flat_hash_map.h"
+#include "src/ir/storage.h"
+#include "src/ir/types/type.h"
+
+namespace raksha::ir {
+
+class Operation;
+class Operator;
+class Value;
+
+namespace value {
+// TODO: Add constructors and make fields private.
+// Also validate fields.
+
+
+// A value that is an argument to an operation.
+class OperationArgument {
+ public:
+  const Operation* operation_;
+  std::string operand_name_;
+};
```

Comment:
You mentioned on the phone that you're uncertain whether this should be a separate base class of Operation. I share that skepticisim; I think we will probably end up wanting to remove this class.

Reply:
Removed this class altogether.